port/unix/omrsysinfo.c

```diff
@@ -3617,6 +3655,63 @@ readCgroupSubsystemFile(struct OMRPortLibrary *portLibrary, uint64_t subsystemFl
 	return rc;
 }
 
+/** 
+ * Checks if the process is running inside docker container 
+ *
+ * @param[in] portLibrary pointer to OMRPortLibrary
+ * @param[out] inDocker pointer to BOOLEAN which on successful return indicates if the process is running in docker container or not 
```

Comment:
Why not return `TRUE` if running in a container and `FALSE` for not / errors?  

The current approach means user code needs to check two things:
```
  BOOLEAN inDocker = false;
   if (0 == isRunningInDocker(portLib, &inDocker)) {
      if (inDocker == TRUE) {
        /* take action */
      }
    }
```
Is there a meaningful distinction between error and not in docker?

Reply:
I am inclined to make this distinction between error and not in docker.
Its caller `omrsysinfo_cgroup_is_system_available` should not attempt to do any processing if any error is encountered.
If we treat error case same as 'not in docker' case, then we may end up in trouble if the error happened when its actually running in docker. In such case it would attempt to read non-existing cgroup control files for determining resource limits.
To be on safer side, if any error occurs, better to not use cgroup limits at all and fallback to regular behavior.